src/Symfony/Component/PasswordHasher/Hasher/UserPasswordHasher.php

```diff
@@ -50,7 +50,7 @@ public function hashPassword($user, string $plainPassword): string
         } elseif ($user instanceof UserInterface) {
             $salt = $user->getSalt();
 
-            if (null !== $salt) {
+            if ($salt) {
```

Comment:
`getSalt()` is typehinted as `@return string`, so `null` would be invalid. This check is now also in sync with the same deprecation check in `CheckCredentialsListener`.